### PR TITLE
Improve shellhook

### PR DESCRIPTION
**Pull request by adnelson**
@ns-gbernsleone this will hopefully fix the issue you were having.

### CODE REVIEW

**Comment by adnelson** [comment]
@ns-gbernsleone I rebased this

File: nodeLib/buildNodePackage.nix [hunk 1]
```diff
@@ -536,29 +536,43 @@ let
         export NODE_PATH=$NODE_MODULES:$NODE_PATH
         # Check if the current directory contains the package.json for
         # this package.
-        if python -c "import json; assert json.load(open('package.json'))['name'] == '$fullName'" 2>/dev/null; then
+        py_cmd='import json; print(json.load(open("package.json"))["name"])'
+        if [[ -e package.json ]] && \
+            [[ $(python -c "$py_cmd" 2>/dev/null) == "$fullName" ]]; then
```
Comment (ns-gbernsleone) [inline nodeLib/buildNodePackage.nix:541]: I can't tell where $fullName gets defined, but it must work because it was in the script before.

**Review by ns-gbernsleone — APPROVED** [review]
LGTM

File: nodeLib/buildNodePackage.nix (hunk 1 above)
Comment (adnelson) [reply nodeLib/buildNodePackage.nix:541]: Yeah fullName is passed in as an argument to the mkDerivation function

**Pull request merged**